src/lib/actions/RunAction.ts

```diff
@@ -53,9 +61,10 @@ export class RunAction {
 			workspaceFiles: input.workspace,
 			pathStartPoints: input['path-start']
 		};
-
+		this.dependencies.progressListeners.forEach(listener => listener.listen(core, ruleSelection));
+		this.dependencies.logEventListeners.forEach(listener => listener.listen(core));
```

Comment:
For the main LogEvent, I would start listening before the engines are created and before rule selection so that we listen to events that might fire there as well.

I guess the difficulty is that you are using the ruleSelection in your listener. Can you avoid doing that?

Reply:
The progress listener is the one that uses the `RuleSelection`. The log listener's don't need it. So I could move the `listen` call for the log event listeners earlier in the process.